easybuild/framework/easyblock.py

```diff
@@ -2855,6 +2857,10 @@ def build_and_install_one(ecdict, init_env):
     start_time = time.time()
     try:
         run_test_cases = not build_option('skip_test_cases') and app.cfg['tests']
+        if not dry_run:
+            # create our reproducability files before carrying out the easyblock steps
+            reprod_dir_root = os.path.dirname(app.logfile)
+            reprod_dir = reproduce_build(app, reprod_dir_root)
```

Comment:
This will definitely impact how the archived easyconfig looks, with many parameters introduced by the easyblock steps no longer recorded in the dump (which is a good thing). Also recording the hooks should sidestep *most* problems with the fact that we also would no longer record any of the changes made by the hooks in the easyblock steps.